### PR TITLE
Rework bookkeeping constants

### LINKED ISSUE #177: Should we swap testVersion and TestVersion?
I think it would be nice if the solutions don't export TestVersion so that golint doesn't complain.

### PR DESCRIPTION
**Pull request by kytrinyx**
I'd like to be able to encourage people to run golint without
forcing them to write a comment for the test version constant.

Closes #177

### CODE REVIEW

File: exercises/circular-buffer/circular_buffer_test.go [hunk 1]
```diff
@@ -19,11 +19,11 @@ import (
 	"testing"
 )
 
-const testVersion = 2
+const targetTestVersion = 3
 
-func TestTestVersion(t *testing.T) {
-	if TestVersion != testVersion {
-		t.Errorf("Found TestVersion = %v, want %v.", TestVersion, testVersion)
+func TesttestVersion(t *testing.T) {
```
Comment (petertseng) [inline exercises/circular-buffer/circular_buffer_test.go]: it seems to make sense to keep this as `testTestVersion` but I don't feel strongly either way

File: exercises/custom-set/custom_set_test.go [hunk 2]
```diff
@@ -33,11 +33,11 @@ import (
 	"testing"
 )
 
-const testVersion = 2
+const targetTestVersion = 3
 
-func TestTestVersion(t *testing.T) {
-	if TestVersion != testVersion {
-		t.Fatalf("Found TestVersion = %v, want %v", TestVersion, testVersion)
+func TesttestVersion(t *testing.T) {
```
Comment (petertseng) [inline exercises/custom-set/custom_set_test.go]: same about `TestTestVersion`

File: exercises/error-handling/error_handling_test.go [hunk 3]
```diff
@@ -10,10 +10,10 @@ import (
 // (in all cases). The function should properly handle errors,
 // as defined by the expectations of this test suite.
 //
-// Also define an exported TestVersion with a value that matches
-// the internal testVersion here.
+// Also define an exported testVersion with a value that matches
+// the internal targetTestVersion here.
 
-const testVersion = 1
+const targetTestVersion = 2
```
Comment (petertseng) [inline exercises/error-handling/error_handling_test.go]: the fact that this is the only mention of `testVersion` in this file indicates to me that this is never being tested, probably good to put in a follow up

File: exercises/food-chain/food_chain_test.go [hunk 4]
```diff
@@ -5,11 +5,11 @@ import (
 	"testing"
 )
 
-const testVersion = 1
+const targetTestVersion = 2
 
-func TestTestVersion(t *testing.T) {
-	if TestVersion != testVersion {
-		t.Errorf("Found TestVersion = %v, want %v.", TestVersion, testVersion)
+func TesttestVersion(t *testing.T) {
```
Comment (petertseng) [inline exercises/food-chain/food_chain_test.go]: another potential `TestTestVersion`

File: exercises/react/react_test.go [hunk 5]
```diff
@@ -8,19 +8,19 @@ import (
...
 // This is a compile time check to see if you've properly implemented New().
 var _ Reactor = New()
 
-// If this test fails and you've proprly defined TestVersion the requirements
+// If this test fails and you've proprly defined testVersion the requirements
 // of the tests have changed since you wrote your submission.
-func TestTestVersion(t *testing.T) {
-	if TestVersion != testVersion {
-		t.Fatalf("Found TestVersion = %v, want %v", TestVersion, testVersion)
+func TesttestVersion(t *testing.T) {
```
Comment (petertseng) [inline exercises/react/react_test.go]: potential `TestTestVersion`

File: exercises/tournament/tournament_test.go [hunk 6]
```diff
@@ -12,10 +12,10 @@ import (
 // Note that unlike other tracks the Go version of the tally function
 // should not ignore errors. It's not idiomatic Go to ignore errors.
 //
-// Also define an exported TestVersion with a value that matches
-// the internal testVersion here.
+// Also define an exported testVersion with a value that matches
```
Comment (petertseng) [inline exercises/tournament/tournament_test.go]: > define an exported `testVersion`

`testVersion` isn't exported

File: exercises/tree-building/tree_test.go [hunk 7]
```diff
@@ -11,10 +11,10 @@ import (
 // where Record is a struct containing int fields ID and Parent
 // and Node is a struct containing int field ID and []*Node field Children.
 //
-// Also define an exported TestVersion with a value that matches
-// the internal testVersion here.
+// Also define an exported testVersion with a value that matches
```
Comment (petertseng) [inline exercises/tree-building/tree_test.go]: `testVersion` isn't exported

File: exercises/triangle/triangle_test.go [hunk 8]
```diff
@@ -84,9 +84,9 @@ func TestKind(t *testing.T) {
 	}
 }
 
-func TestTestVersion(t *testing.T) {
-	if TestVersion != testVersion {
-		t.Fatalf("Found TestVersion = %v, want %v", TestVersion, testVersion)
+func TesttestVersion(t *testing.T) {
```
Comment (petertseng) [inline exercises/triangle/triangle_test.go]: potential `TestTestVersion`

File: exercises/word-search/word_search_test.go [hunk 9]
```diff
@@ -7,10 +7,10 @@ import (
 
 // Define a function Solve(words []string, puzzle []string) (map[string][2][2]int, error).
 //
-// Also define an exported TestVersion with a value that matches
-// the internal testVersion here.
+// Also define an exported testVersion with a value that matches
```
Comment (petertseng) [inline exercises/word-search/word_search_test.go]: `testVersion` isn't exported

File: exercises/error-handling/error_handling_test.go [hunk 10]
```diff
@@ -10,10 +10,10 @@ import (
 // (in all cases). The function should properly handle errors,
 // as defined by the expectations of this test suite.
 //
-// Also define an exported TestVersion with a value that matches
-// the internal testVersion here.
+// Also define an exported testVersion with a value that matches
```
Comment (petertseng) [inline exercises/error-handling/error_handling_test.go]: `testVersion` isn't exported, so the instruction in the comment is inaccurate

File: exercises/hello-world/hello_test.go [hunk 11]
```diff
@@ -4,10 +4,10 @@ import "testing"
 
 // Define a function HelloWorld(string) string.
 //
-// Also define an exported TestVersion with a value that matches
-// the internal testVersion here.
+// Also define an exported testVersion with a value that matches
```
Comment (petertseng) [inline exercises/hello-world/hello_test.go]: `testVersion` isn't exported, so the instruction in the comment is inaccurate

File: exercises/leap/leap_test.go [hunk 12]
```diff
@@ -4,14 +4,14 @@ import "testing"
 
 // Define a function IsLeapYear(int) bool.
 //
-// Also define an exported TestVersion with a value that matches
-// the internal testVersion here.
+// Also define an exported testVersion with a value that matches
```
Comment (petertseng) [inline exercises/leap/leap_test.go]: `testVersion` isn't exported, so the instruction in the comment is inaccurate

File: exercises/poker/poker_test.go [hunk 13]
```diff
@@ -7,10 +7,10 @@ import (
 
 // Define a function BestHand([]string) ([]string, error).
 //
-// Also define an exported TestVersion with a value that matches
-// the internal testVersion here.
+// Also define an exported testVersion with a value that matches
```
Comment (petertseng) [inline exercises/poker/poker_test.go]: `testVersion` isn't exported, so the instruction in the comment is inaccurate

File: exercises/react/react_test.go [hunk 14]
```diff
@@ -8,19 +8,19 @@ import (
 // Define a function New() Reactor and the stuff that follows from
 // implementing Reactor.
 //
-// Also define an exported TestVersion with a value that matches
-// the internal testVersion here.
+// Also define an exported testVersion with a value that matches
```
Comment (petertseng) [inline exercises/react/react_test.go]: `testVersion` isn't exported, so the instruction in the comment is inaccurate

**Comment by petertseng** [comment]
Thanks for dealing with the mechanical but "somebody's gotta do it" change. My comments fall into three categories:

* Sure about `TesttestVersion` instead of `TestTestVersion`? The former seems to imply the function is named "testtest version" instead of "test test version" by the rules of camel casing. I don't feel strongly about this.
* The instructions of "Also define an exported `TestVersion`" has changed to "Also define an exported `testVersion`" but this instruction is inaccurate since `testVersion` isn't exported.
* error-handling exercise doesn't seem to check its test version at all.

File: exercises/tree-building/tree_test.go (hunk 7 above)
Comment (kytrinyx) [inline exercises/tree-building/tree_test.go]: Ah, thanks for catching that!

File: exercises/triangle/triangle_test.go (hunk 8 above)
Comment (kytrinyx) [inline exercises/triangle/triangle_test.go]: Oops. Yepp, fixing.

**Comment by kytrinyx** [comment]
I totally missed the "exported" comment. I changed the stuff in the README, but completely overlooked the comments in the test files themselves.

The `Testtest` was also a mistake. I did some global renames and didn't notice it in the diff.

I'll put in a PR for the test version on the error handling thing in a moment.

**Comment by petertseng** [comment]
:+1:

**Pull request merged**